crates/turborepo-cache/src/http.rs

```diff
@@ -68,20 +79,37 @@ impl HTTPCache {
     ) -> Result<(), CacheError> {
         let mut artifact_body = Vec::new();
         self.write(&mut artifact_body, anchor, files).await?;
+        let bytes = artifact_body.len();
 
         let tag = self
             .signer_verifier
             .as_ref()
             .map(|signer| signer.generate_tag(hash.as_bytes(), &artifact_body))
             .transpose()?;
 
+        let stream = tokio_util::codec::FramedRead::new(
```

Comment:
To watch the upload rate we need to use a stream. `reqwest` only supports a Stream of Result<Bytes, Error> so that it is compatible with other IO errors. We compose a cursor with a framed read to do this.